### DISCUSSION TITLE
‘No default value’ for primary key column Relationship Picker

### DISCUSSION THREAD

**Discussion by mmalessa80** [Budibase/budibase#6651] [Help]
I have a form for T1 with a relationship picker to T2 and a Relationship table with T1_fk, T2_fk, and id (primary key) columns. When clicking ‘save’ it saves the row to T1, however then I get an error for ‘No default value for column id’.

I have been unable to find where to set the default value for my primary key which is just supposed to be Concatenate(T1_fk,T2_fk). I would appreciate any help with this.

**Comment by melohagan** [comment]
As mentioned in this issue (https://github.com/Budibase/budibase/discussions/6567) composite keys are not yet supported. 

I don't think setting a default primary key value would be done in Budibase. This StackOverflow post might help? https://stackoverflow.com/questions/25865104/field-id-doesnt-have-a-default-value

**Reply by mmalessa80** [reply]
Thank you for the SO suggestion. To test the composite key limitation within Budibase I made some changes in my MySQL database by dropping the primary key constraint (column name 'id'), then dropping the column with name 'id', and then adding a composite primary key (T1_fk,T2_fk). Now my bridge table configuration with my DB is as follows:

![image](https://user-images.githubusercontent.com/106758851/179563656-d56945d0-c790-4311-b1b7-213901afbd0e.png)

Within Budibase I went to my MySQL datasource and clicked the 'Fetch Tables' button again and the tables now reflect the following:

![image](https://user-images.githubusercontent.com/106758851/179564036-1fe343d5-226c-48cf-a9be-d8e9e3b98389.png)

So far, so good. However, if I go to the datasource table for this bridge table, it does not fetch any rows within Budibase, and no matter what I try I cannot delete the 'ID' column within Budibase, although I had dropped that column in the MySQL table. See the result:

![image](https://user-images.githubusercontent.com/106758851/179564749-094d1f20-faa9-4d91-9dd8-ca3e46bf6b8c.png)

Now having made these adjustments, I went back to my form for a new entry for AnnualPermitData and set the default value for PermitList relationship picker to the following:

![image](https://user-images.githubusercontent.com/106758851/179565876-f9ffbe33-91d3-4a57-b537-4808132bdc3a.png)

This is taking the result of a query that I run as part of a process and just mapping the primary keys of the PermitList table that I want to assign to this AnnualPermitData new row record in the format of ['abcd1234','ertw3412']. When I click 'save' the new row shows up in the AnnualPermitData table with the correct relationship to PermitList as seen below:

![image](https://user-images.githubusercontent.com/106758851/179568576-8fd97752-f5b3-4bd3-ba38-aa228b9d90fb.png)

So I suppose in that sense, the fact that the bridge table in Budibase does not reflect data actually stored in the MySQL table certainly might present a problem, however it appears that Budibase is in fact keeping track of the relationship, albeit I don't know how. Any further clarification that you might be able to provide on this topic would certainly be welcome. I apologize for the long winded response.

**Reply by melohagan** [reply]
> However, if I go to the datasource table for this bridge table, it does not fetch any rows within Budibase, and no matter what I try I cannot delete the 'ID' column within Budibase, although I had dropped that column in the MySQL table.

This may be related to this bug: https://github.com/Budibase/budibase/issues/6737

 > it appears that Budibase is in fact keeping track of the relationship, albeit I don't know how. Any further clarification that you might be able to provide on this topic would certainly be welcome.

I've updated the docs to show an example of a many to many relationship: https://docs.budibase.com/docs/postgresql#many-to-many
